### ISSUE TITLE
Thermalization of CPU LB broken

### ISSUE THREAD

**Issue by pkreissl** [espressomd/espresso#3804]
IMHO broken for CPU. Minimal example:
```python
import argparse

parser = argparse.ArgumentParser()
parser.add_argument("--use_gpu", default=False, action="store_true")
parser.add_argument("--kt", type=float, default=0)
args = parser.parse_args()

import espressomd
from espressomd.observables import LBFluidPressureTensor

system = espressomd.System(box_l=[20, 20, 20])
system.time_step = 0.01
system.cell_system.skin = 0.4
if args.use_gpu:
    lbf = espressomd.lb.LBFluidGPU(agrid=1.0, dens=1.9, visc=1.0, tau=0.01, kT=args.kt, seed=7)
else:
    lbf = espressomd.lb.LBFluid(agrid=1.0, dens=1.9, visc=1.0, tau=0.01, kT=args.kt, seed=7)
system.actors.add(lbf)
lb_pressure = LBFluidPressureTensor()
for i in range(2):  # change as you like
    system.integrator.run(1000)  # same here, problem persists
    print(lb_pressure.calculate())
```
Testing via `run_minimal.sh`:
```bash
#!/bin/bash

echo "CPU, kt=0:"
./pypresso minimal.py
echo "GPU, kt=0:"
./pypresso minimal.py --use_gpu
echo "CPU, kt=1:"
./pypresso minimal.py --kt 1
echo "GPU, kt=1:"
./pypresso minimal.py --use_gpu --kt 1
```
yields:
```
$ ./run_minimal.sh 
CPU, kt=0:
[[6333.33325386    0.            0.        ]
 [   0.         6333.33325386    0.        ]
 [   0.            0.         6333.33325386]]
[[6333.33325386    0.            0.        ]
 [   0.         6333.33325386    0.        ]
 [   0.            0.         6333.33325386]]
GPU, kt=0:
WARNING: More than one GPU detected, please note ESPResSo uses device 0 by default regardless of usage or capability. The GPU to be used can be modified by setting System.cuda_init_handle.device.
[[6333.33353698    0.            0.        ]
 [   0.         6333.33353698    0.        ]
 [   0.            0.         6333.33353698]]
[[6333.33353698    0.            0.        ]
 [   0.         6333.33353698    0.        ]
 [   0.            0.         6333.33353698]]
CPU, kt=1:
[[6.39378045e+03 1.90899139e+00 1.95662446e+00]
 [1.90899139e+00 6.38988958e+03 1.94024253e+00]
 [1.95662446e+00 1.94024253e+00 6.38844944e+03]]
[[6.39384017e+03 1.91773972e+00 2.03000427e+00]
 [1.91773972e+00 6.39001967e+03 1.98502662e+00]
 [2.03000427e+00 1.98502662e+00 6.38863714e+03]]
GPU, kt=1:
WARNING: More than one GPU detected, please note ESPResSo uses device 0 by default regardless of usage or capability. The GPU to be used can be modified by setting System.cuda_init_handle.device.
[[ 6.33442971e+03 -2.38058610e-02 -6.62311537e-02]
 [-2.38058610e-02  6.33435239e+03  4.31334905e-02]
 [-6.62311537e-02  4.31334905e-02  6.33439990e+03]]
[[ 6.33472237e+03 -5.51996531e-02 -2.82031257e-02]
 [-5.51996531e-02  6.33468426e+03  1.38701318e-02]
 [-2.82031257e-02  1.38701318e-02  6.33475568e+03]]
```
`kT=0` case is fine, but with thermalization, CPU produces off-dagonal pressure values of order $10^0$, instead of GPU, order $10^{-2}$ which seems more reasonable, I think.

**Comment by RudolfWeeber** [comment]
Can you please check in Ulf Schillers PhD thesis what are the expectation values for

<sigma_{ij}^2> for thermalized lb?

 

What are the integrals of the stress acf for cpu and gpu, respectively?

**Comment by pkreissl** [comment]
With gpu the acf integral produces ~ resonable values for the viscosity via Green-Kubo, for cpu, the acf does not even decay yet (~1e6 ~steps~ samples, runs A LOT slower than gpu...)

**Comment by pkreissl** [comment]
Concerning Ulfs PhD thesis, I have not yet had to much to do with it, apart from browsing a bit and I am thus not immediately familiar with the different notations and definitions used there. Will give it a try, however, it will probably be a lot faster, if someone more involved with this, e.g. @mkuron or @KaiSzuttor could comment...

**Comment by pkreissl** [comment]
ACF of off-diagonal elements using 400000 samles (and a slightly differen lb parameter set, however nothing fancy), evaluated every second time step, for GPU:
![Screenshot from 2020-07-17 13-49-50](https://user-images.githubusercontent.com/12623404/87783353-93b21e80-c834-11ea-96d1-59cf3a95026e.png)
and for CPU:
![Screenshot from 2020-07-17 13-50-35](https://user-images.githubusercontent.com/12623404/87783356-944ab500-c834-11ea-9b1f-af3ce9c93919.png)

**Comment by pkreissl** [comment]
In offline discussion, @RudolfWeeber suggested that this issue might be a regression. I checked for ESPResSo version 4.1.0, same problem, for 4.0.0 there doesn't seem to be a fluid stress observable (neither `LBFluidPressureTensor` nor `LBFluidStress`)?! So this problem does indeed seem to have been around for some time.

**Comment by KaiSzuttor** [comment]
How do you know that the issue has been around for some time if you cannot check for versions older than 4.1.0 (which is less than a year old)?

**Comment by pkreissl** [comment]
> How do you know that the issue has been around for some time if you cannot check for versions older than 4.1.0 (which is less than a year old)?

Let me rephrase that: ~I couldn't find the feature before 4.1.0 (if it had just a different name, please do tell me, I'll check for the issue)~ The `LBFluidStress` observable was introduced with 4.1.0 (PR #2054), so it seems to have been broken from the beginning...
